### DISCUSSION TITLE
vips jp2 output is sometimes raw JPEG2000 stream instead of proper jp2

### DISCUSSION THREAD

**Discussion by jrochkind** [libvips/libvips#3597] [Q&A] [answered]
OK, I use `vipsthumbnail` to convert a TIFF to a JP2. 

This works fine on my development MacOS box. It works fine on my deployment (to heroku). 

It weirdly works _differently_ in _Github CI_, where the process is outputting a raw "JPEG2000 codestream" instead of a proper JP2 file with metadata. 

I noticed this when it turned out one of the tools i was using couldn't handle a raw JPEG2000 codestream -- but it was failing only in my test suite run on Github CI, and I couldn't reproduce locally or on my deployment machine. 

But I have now verified. Perhaps it is due to different vips (or dependencies like ImageMagick) versions? I wonder if this is a known bug fixed at some point?  I have not been able to find it in changelog. 

My test reproduction:

    $ vipsthumbnail input.tiff -o output.jp2[Q=40,subsample-mode=off]
    $ file output.jp2
    # returns either `JPEG 2000 Part 1 (JP2)` (GOOD), or `JPEG 2000 codestream` (BAD)

* My MacOS, with a brew-installed vips 8.14.2 and ImageMagick 7.1.1 => GOOD, `JPEG 2000 Part 1 (JP2)`
* Github CI, running Ubuntu 22.04.2, with an apt-installed vips 8.12.1,  ImageMagick 6.9.11-60 => BAD `JPEG 2000 codestream`
* Heroku-22, running on Ubuntu 22.04, but with a custom install of vips 8.13.0 via https://github.com/brandoncc/heroku-buildpack-vips => GOOD `JPEG 2000 Part 1 (JP2)`

Any ideas why one of these hosts, with the identical command, might be producing JPEG2000 codestream output instead of proper JP2 output?  Is it simply a bug that has been fixed since 8.12.1?

**Comment by jrochkind** [comment]
The TIFF file I'm using as input, in case it matters: https://scih-data-dev.s3.amazonaws.com/jrochkind-share/ORIG_TEST.tiff

**Comment by jrochkind** [comment]
Any thoughts @jcupitt , should I file this as a bug in Issues?

**Comment by jcupitt** [comment, marked as answer]
Ooop, sorry @jrochkind, I was sure I answered this, but I was thinking of another very similar issue. This is a bug that was fixed in 8.14.

**Reply by jrochkind** [reply]
perfect, thanks!
